duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt

```diff
@@ -305,15 +320,30 @@ class RealDuckPlayer @Inject constructor(
             return processDuckPlayerUri(url, webView)
         } else {
             if (!isFeatureEnabled) return null
+            val webViewUrl = withContext(dispatchers.main()) { webView.url }
             if (isYoutubeWatchUrl(url)) {
                 return processYouTubeWatchUri(request, url, webView)
             } else if (isSimulatedYoutubeNoCookie(url)) {
                 return processSimulatedYouTubeNoCookieUri(url, webView)
+            } else if (duckPlayerFeature.addCustomEmbedReferer().isEnabled() && isYouTubeNoCookieEmbedUri(url, webViewUrl)) {
+                return try {
+                    WebResourceResponse("text/html", "UTF-8", getEmbedWithReferer(request))
```

Comment:
getEmbedWithReferer() can return null, would that be the correct thing to send in the WebResourceResponse?

Reply:
Good call. When I made it nullable, I was thinking to return a null response (so we continue loading the original request), but that's not what I'm doing here. Thanks for catching it!